pytest.ini

```diff
@@ -2,3 +2,10 @@
 markers =
     serial: tests that cannot be reliably ran with pytest multiprocessing
     timeout: used with pytest-timeout
+addopts =
+    -r a
+    --color yes
+    --showlocals
+    --verbose
+    --numprocesses auto
```

Comment:
Testing your branch, and I want to voice a preference that we not put this in the pytest config, because running with multiprocessing makes debugging much harder. Options like `-s` are unusable.

Reply:
It also makes all the test commands fail for me for reasons I don't fully understand yet. Maybe how my virtual environment carries over into the subprocesses, but not sure

Reply:
My aim with adding these options was to have `pytest` run the same locally as it does in CI. But I have also felt the friction this adds. It's possible to disable multiprocessing with `tox -e py3 -- -n0`, but it's a bit of a pain.

I actually like the way `ansible-test` does this more dynamically with a cap, but `tox` doesn't have a way to do that easily (it can be done with a rather ungainly `shell` command, but I don't care for that approach).

I also want to move to using `--forked` for tests as well for tests that are not thread-safe. I'm hoping that will remove the need for the `-m serial` and `-m "not serial"` commands, but I'm not 100% sure why those are needed.

Anyway, what do you think about being able to do `-- -n0` locally in order to have the default behavior be the same locally and in CI?

Reply:
Yeah, I've come around to see the wisdom of `pytest.ini` keeping local runs synced with how CI works.

The main issue I had was that tests were failing due to the bug we've already discussed here. Solution there is to fix the bug! Running locally, `-n0` doesn't seem like too much hassle to expect of people.

Reply:
Note that if you use `--pdb` it also tends to seamlessly turn the runs into `-s -n0`.

Reply:
Alright, I'll work on fixing the actual bug. I do love how `--pdb` and `-n auto` play nicely together.